api_app/analyzers_manager/observable_analyzers/bgp_ranking.py

```diff
@@ -0,0 +1,102 @@
+import json
+import logging
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class BGPRanking(classes.ObservableAnalyzer):
+    """
+    wrapper for https://github.com/D4-project/BGP-Ranking
+    """
+
+    observable_name: str
+    period: int  # optional
+
+    def update(self) -> bool:
+        pass
+
+    def run(self):
+        logger.info("Running BGP_Ranking")
+        urls = {
+            "getASN": "https://bgpranking-ng.circl.lu/ipasn_history/?ip=",
+            "getASNRank": "https://bgpranking-ng.circl.lu/json/asn",
+            "getASNHistory": "https://bgpranking-ng.circl.lu/json/asn_history",
+        }
+        finalresposne = {}
+
+        # get ASN from ip
+        try:
+            response = requests.get(urls["getASN"] + self.observable_name)
+            response.raise_for_status()
+            response = response.json()
+            finalresposne["asn"] = response["response"][
+                list(response["response"].keys())[0]
+            ]["asn"]
+            logger.info("ASN extracted from IP")
+
+            # get ASN rank from extracted ASN
+
+            response = requests.post(
+                urls["getASNRank"], data=json.dumps({"asn": finalresposne["asn"]})
+            )
+            response.raise_for_status()
+            response = response.json()
+            finalresposne["asn_description"] = response["response"]["asn_description"]
+            finalresposne["asn_rank"] = response["response"]["ranking"]["rank"]
+            finalresposne["asn_position"] = response["response"]["ranking"]["position"]
+            logger.info("ASN rank and position extracted from ASN")
+
+            if self.period:
+                # get ASN history from extracted ASN
+                response = requests.post(
+                    urls["getASNHistory"],
+                    data=json.dumps(
+                        {"asn": finalresposne["asn"], "period": self.period}
+                    ),
+                )
+                response.raise_for_status()
+                response = response.json()
+                finalresposne["asn_history"] = response["response"]["asn_history"]
+                logger.info("ASN history extracted from ASN")
+
+        except (
+            requests.exceptions.RequestException,
+            TypeError,
+            json.JSONDecodeError,
+            KeyError,
+            AttributeError,
+        ) as e:
+            # Handle various specific exceptions
+            logger.error(f"Exception: {e}")
+            raise AnalyzerRunException(f"AnalyzerRunException: {e}")
```

Comment:
I **hate** a try catch that big that pretty much manage every single line of execution.
Moreover, if in the catch branch we are not doing actually anything.
Imho we can just remove the try catch completely and let the `start` function manage this accordingly

Reply:
okay, I see. Please correct me if I'm wrong here, I need to get rid of the error handling in the module entirely, right? by start function you mean `run` right?
> Moreover, if in the catch branch we are not doing actually anything.
Is it okay to not log and raise an exception? I'm a bit confused here...

Reply:
> I need to get rid of the error handling in the module entirely, right?
right
> by start function you mean run right?
Nope, i mean the `start` function. Its code is in the `plugin` parent class.
>Is it okay to not log and raise an exception? I'm a bit confused here...
If you check the code of the `start` function you will see that we are already doing exactly that(and else) in the `after_run_failed` error handler function

Reply:
done.